lumispy/tests/utils/test_plot_span_map.py

```diff
@@ -0,0 +1,158 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019-2022 The LumiSpy developers
+#
+# This file is part of LumiSpy.
+#
+# LumiSpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the license, or
+# (at your option) any later version.
+#
+# LumiSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with LumiSpy. If not, see <https://www.gnu.org/licenses/#GPL>.
+
+from functools import wraps
+
+import pytest
+import numpy as np
+
+import hyperspy.api as hs
+
+from lumispy.utils.analysis import plot_span_map
+
+
+@pytest.fixture
+def test_signal():
+    test_data = np.zeros((64, 64, 1024))
+    test_data[32:, :, 300:] = 1
+    test_data[48:, :, 500:] = 2
+    sig = hs.signals.Signal1D(
+        test_data,
+        axes=[
+            {"name": "x", "size": 64, "offset": 0, "scale": 1, "units": "um"},
+            {"name": "y", "size": 64, "offset": 0, "scale": 1, "units": "um"},
+            {
+                "name": "Wavelength",
+                "size": 1024,
+                "offset": 0,
+                "scale": 1,
+                "units": "nm",
+            },
+        ],
+    )
+
+    sig.set_signal_type("CLSEM")
+    return sig
+
+
+def sig_mpl_compare(type: set(["sig", "nav"])):
+    def wrapper(f):
+        @pytest.mark.mpl_image_compare(baseline_dir="plot_span_map")
```

Comment:
Thanks for the comments Jordi!

I am trying to follow the convention set out by hyperspy i.g. for testing the drawing library they make use of these kinds of tests:

https://github.com/hyperspy/hyperspy/tree/RELEASE_next_minor/hyperspy/tests/drawing/plot_model1d

My concern is with the interactivity, I think it would be good to ensure that the appearance of the plots updates as the spans are moved around, which I think the current tests do.